contracts/attesters/hydra-s1/HydraS1AccountboundAttester.sol

```diff
@@ -0,0 +1,289 @@
+    MANDATORY FUNCTIONS TO OVERRIDE FROM ATTESTER.SOL
+  *******************************************************/
+
+  /**
+   * @dev Returns the actual attestations constructed from the user request
+   * @param request users request. Claim of having an account part of a group of accounts
+   * @param proofData snark public input as well as snark proof
+   */
+  function buildAttestations(Request calldata request, bytes calldata proofData)
+    public
+    view
+    virtual
+    override(IAttester, HydraS1SimpleAttester)
+    returns (Attestation[] memory)
+  {
+    Attestation[] memory attestations = super.buildAttestations(request, proofData);
+
+    uint256 nullifier = proofData._getNullifier();
+    attestations[0].extraData = abi.encodePacked(
+      attestations[0].extraData,
+      generateAccountboundExtraData(nullifier, attestations[0].owner)
+    );
+
+    return (attestations);
+  }
+
+  /*******************************************************
+    OPTIONAL HOOK VIRTUAL FUNCTIONS FROM ATTESTER.SOL
+  *******************************************************/
+  /**
+   * @dev Hook run before recording the attestation.
+   * Throws if nullifier already used, not a renewal, and nullifier on cooldown.
+   * @param request users request. Claim of having an account part of a group of accounts
+   * @param proofData provided to back the request. snark input and snark proof
+   */
+  function _beforeRecordAttestations(Request calldata request, bytes calldata proofData)
+    internal
+    virtual
+    override
+  {
+    uint256 nullifier = proofData._getNullifier();
+    address previousNullifierDestination = _getDestinationOfNullifier(nullifier);
+
+    HydraS1Claim memory claim = request._claim();
+
+    // check if the nullifier has already been used previously, if so it may be on cooldown
+    if (
+      previousNullifierDestination != address(0) &&
+      previousNullifierDestination != claim.destination
+    ) {
+      uint32 cooldownDuration = _getCooldownDurationForGroupIndex(claim.groupProperties.groupIndex);
+      if (cooldownDuration == 0) {
+        revert CooldownDurationNotSetForGroupIndex(claim.groupProperties.groupIndex);
+      }
+      if (_isOnCooldown(nullifier, cooldownDuration)) {
+        uint16 burnCount = _getNullifierBurnCount(nullifier);
+        revert NullifierOnCooldown(
+          nullifier,
+          previousNullifierDestination,
+          burnCount,
+          cooldownDuration
+        );
+      }
+
+      // Delete the old Attestation linked to the nullifier before recording the new one (accountbound behaviour)
+      _deletePreviousAttestation(nullifier, claim, previousNullifierDestination);
+
+      _setNullifierOnCooldownAndIncrementBurnCount(nullifier);
+    }
+    _setDestinationForNullifier(nullifier, request.destination);
+  }
+
+  /*******************************************************
+    LOGIC FUNCTIONS RELATED TO ACCOUNTBOUND BEHAVIOUR
```

Comment:
Be consistent with how you comment sections, in HydraS1Base you update to use. Chose one or the other

```
  /**
   *
   * INITIALIZATION FUNCTIONS
   *
   */
```